client/core/trade.go

```diff
@@ -1239,14 +1245,124 @@ func (t *trackedTrade) isSwappable(ctx context.Context, match *matchTracker) (re
 				"but we have not seen the counter-party's redemption yet. This could just"+
 				" be network latency.", match)
 		}
+
 		match.setSwapConfirms(int64(confs))
+
 		t.notify(newMatchNote(TopicConfirms, "", "", db.Data, t, match))
 		return false, false
 	}
 
 	return false, false
 }
 
+// updateFeesPaid starts a goroutine that updates the fees used for dynamic fee
+// checker transactions. We do not know the exact fees the tx will use until
+// they are mined, so this waits until they are mined and updates the value for
+// the entire trade.
+//
+// TODO: This method does not persist if the client is shut down. It probably
+// should because payed fees will be incorrect forever for this trade if not.
+func (t *trackedTrade) updateFeesPaid(ctx context.Context, match *matchTracker, isInit bool) {
+	wallet := t.wallets.fromWallet
+	if !isInit {
+		wallet = t.wallets.toWallet
+	}
+	feeChecker, dynamic := wallet.Wallet.(asset.DynamicTxFeeChecker)
+	if !dynamic {
+		return
+	}
+	go func() {
+		ctx, cancel := context.WithTimeout(ctx, updateFeesTimeout)
+		defer cancel()
+		txType := "swap"
+		if !isInit {
+			txType = "redemption"
+		}
+		var coinID []byte
+
+		// Check if a swap or redeem coin id has been populated in the
+		// match tracker. If it has we ask the wallet for the fees paid
+		// and add that to either the total swap or redeem fees for the
+		// trade.
+		check := func() (doReturn bool) {
+			t.mtx.RLock()
+			if isInit {
+				coinID = []byte(match.MetaData.Proof.MakerSwap)
+				if match.Side != order.Maker {
+					coinID = []byte(match.MetaData.Proof.TakerSwap)
+				}
+			} else {
+				coinID = []byte(match.MetaData.Proof.MakerRedeem)
+				if match.Side != order.Maker {
+					coinID = []byte(match.MetaData.Proof.TakerRedeem)
+				}
+			}
+			contractData := match.MetaData.Proof.ContractData
+			t.mtx.RUnlock()
+			if coinID == nil {
+				return false
+			}
+			confs, err := wallet.RegFeeConfirmations(ctx, coinID)
+			if err != nil {
+				if errors.Is(err, asset.CoinNotFoundError) {
+					return false
+				}
+				t.dc.log.Errorf("Failed to determine actual %s transaction fees paid for "+
+					"match %s: %v", txType, match, err)
+				return true
+			}
+			if confs < updateFeesConfs {
+				return false
+			}
+			actualSwapFees, _, position, err := feeChecker.TransactionFeesPaid(ctx, coinID, contractData, isInit)
```

Comment:
It works like this, but why are you returning both the position and the list of secret hashes?

Reply:
It was suggested to do so, but I wasn't sure why. Can remove the secrets return. Coin id is the secret here, so could also go with just the secrets. Probably that was the thinking.

Reply:
It was suggested to just return the secret hashes and let the caller match with the known `MatchProof.SecretHash`, but not also the position.  However, just the position does seem to be fine, so yeah let's ditch the secrets return.

Also, it's a strangely named method now that it only works on swap contract interactions and not all txns.  I feel like we should rename it or have it work in the absence of provided contract data.

Finally, I'm wondering if init vs redeem cannot be determined from the tx call data instead of requiring that `isInit`.

Reply:
Yeah I think just submitting a CoinID and getting back the list of secret hashes and the fee is possible.

Reply:
> MatchProof.SecretHash, but not also the position.

Oh yeah, that would have been the thing to do.

> Finally, I'm wondering if init vs redeem cannot be determined from the tx call data instead of requiring that isInit.

Can try one then the other and only error if both fail.

> Also, it's a strangely named method 

`TransactionFeesPaid` or `updateFeesPaid`?

Reply:
https://github.com/decred/dcrdex/compare/4ad09b10f27bede4e765edaaa5772927a5ef303e..1eb6260a25e50f9e4fb902322d53ab87757558aa

Decided to just use the secrets so removed the position. It felt weird to just return the position. Caller is checking the returns against the secret hash.

No longer taking the isInit bool.

Added `Dynamic` to everything.

Or did I missread.

>  I feel like we should rename it or have it work in the absence of provided contract data.

It needs the contract data for the version. change to `SwapFeesPaid` ok?

Reply:
renamed things to `Swap` https://github.com/decred/dcrdex/compare/1eb6260a25e50f9e4fb902322d53ab87757558aa..7ea288bb8c61cfdb74190cd34d8c3c36c5370135

Reply:
The core tests wait a bit now without the first check...